# Descriptive analyzes with complete data

In [79]:
import pandas as pd
import networkx as nx
import matplotlib.pyplot as plt
import numpy as np

In [2]:
complete_data=pd.read_csv('data_complete_3_years.csv')
complete_data

exporter importer secgroup product         v2020         v2018  \
0         ALB      ALB      CAT      TT      0.000000      0.000000   
1         ALB      ARG      CAT      TT      0.331972      0.260846   
2         ALB      AUS      CAT      TT      2.099945      5.103496   
3         ALB      AUT      CAT      TT     28.753344     29.996366   
4         ALB      BEL      CAT      TT      2.912183      3.848980   
...       ...      ...      ...     ...           ...           ...   
6395      VNM      UKR      CAT      TT    469.650495    420.582012   
6396      VNM      URY      CAT      TT     61.885624     58.309505   
6397      VNM      USA      CAT      TT  74124.562477  45411.468563   
6398      VNM      VEN      CAT      TT     22.617408     22.516020   
6399      VNM      VNM      CAT      TT      0.000000      0.000000   

             v2019 exporter_name       importer_name  relative_volume_2018  \
0         0.000000       Albania             Albania              0.000000   
1         0.600098       Albania           Argentina              0.000120   
2         3.573215       Albania           Australia              0.002354   
3        35.448914       Albania             Austria              0.013834   
4         5.427032       Albania  Belgium-Luxembourg              0.001775   
...            ...           ...                 ...                   ...   
6395    437.185631      Viet Nam             Ukraine              0.001668   
6396     61.332123      Viet Nam             Uruguay              0.000231   
6397  61879.449603      Viet Nam                 USA              0.180129   
6398     20.980374      Viet Nam           Venezuela              0.000089   
6399      0.000000      Viet Nam            Viet Nam              0.000000   

      relative_volume_2019  relative_volume_2020  
0                 0.000000              0.000000  
1                 0.000271              0.000172  
2                 0.001614              0.001087  
3                 0.016010              0.014887  
4                 0.002451              0.001508  
...                    ...                   ...  
6395              0.001573              0.001550  
6396              0.000221              0.000204  
6397              0.222713              0.244655  
6398              0.000076              0.000075  
6399              0.000000              0.000000  

[6400 rows x 12 columns]

In [3]:
print('Number of exporters' , len(complete_data.exporter_name.unique()) )
print('Number of importers', len(complete_data.importer_name.unique()) )

Number of exporters 80
Number of importers 80


In [4]:
list(complete_data.importer_name.unique()) == list(complete_data.exporter_name.unique())

True

We work with a dataset of 6400 rows and 12 columns, representing trade between 80 contries.

In [6]:
d = {'Country': complete_data.exporter_name.unique()}
By_Country = pd.DataFrame(data=d)

exp_2018 = []
exp_2020 = []

imp_2020 = []
imp_2018 = []

for country in By_Country.Country : 
    exp_2020.append(complete_data.loc[complete_data['exporter_name'] == country]['v2020'].sum())
    exp_2018.append(complete_data.loc[complete_data['exporter_name'] == country]['v2018'].sum())
    
    imp_2020.append(complete_data.loc[complete_data['importer_name'] == country]['v2020'].sum())
    imp_2018.append(complete_data.loc[complete_data['importer_name'] == country]['v2018'].sum())
    
By_Country['exp_2018'] = exp_2018
By_Country['exp_2020'] = exp_2020

By_Country['imp_2020'] = imp_2020
By_Country['imp_2018'] = imp_2018

By_Country['trade_balance_2020'] = By_Country['exp_2020']  - By_Country['imp_2020']
By_Country['trade_balance_2018'] = By_Country['exp_2018']  - By_Country['imp_2018']

By_Country['ratio_exp_2020_2018'] =  By_Country['exp_2020']/By_Country['exp_2018']

By_Country['ratio_imp_2020_2018'] =  By_Country['imp_2020']/By_Country['imp_2018']
By_Country['ratio_balance_2020_2018'] =  By_Country['trade_balance_2020'] / By_Country['trade_balance_2018']

By_Country

Country      exp_2018      exp_2020      imp_2020  \
0              Albania  2.168300e+03  1.931458e+03  5.194407e+03   
1            Argentina  5.766592e+04  5.310376e+04  4.106762e+04   
2            Australia  2.323897e+05  2.290638e+05  1.802182e+05   
3              Austria  1.677247e+05  1.542132e+05  1.576361e+05   
4   Belgium-Luxembourg  2.980566e+05  2.772950e+05  2.876453e+05   
..                 ...           ...           ...           ...   
75             Ukraine  4.740671e+04  4.764214e+04  5.477827e+04   
76             Uruguay  8.365434e+03  7.341809e+03  8.805665e+03   
77                 USA  1.280713e+06  1.099495e+06  2.083720e+06   
78           Venezuela  3.154036e+04  4.405493e+03  7.696720e+03   
79            Viet Nam  2.521053e+05  3.029756e+05  2.630811e+05   

        imp_2018  trade_balance_2020  trade_balance_2018  ratio_exp_2020_2018  \
0   5.348314e+03        -3262.948961        -3180.014601             0.890771   
1   6.197809e+04        12036.144733        -4312.173629             0.920886   
2   2.021444e+05        48845.613274        30245.232612             0.985688   
3   1.754705e+05        -3422.907613        -7745.780413             0.919442   
4   3.281751e+05       -10350.270447       -30118.493714             0.930343   
..           ...                 ...                 ...                  ...   
75  5.814346e+04        -7136.129755       -10736.752416             1.004966   
76  1.141659e+04        -1463.855725        -3051.156121             0.877636   
77  2.274608e+06      -984225.498531      -993895.141399             0.858502   
78  1.475661e+04        -3291.226935        16783.746932             0.139678   
79  2.425580e+05        39894.481927         9547.334307             1.201782   

    ratio_imp_2020_2018  ratio_balance_2020_2018  
0              0.971223                 1.026080  
1              0.662615                -2.791201  
2              0.891532                 1.614986  
3              0.898362                 0.441906  
4              0.876499                 0.343652  
..                  ...                      ...  
75             0.942123                 0.664645  
76             0.771304                 0.479771  
77             0.916079                 0.990271  
78             0.521578                -0.196096  
79             1.084611                 4.178599  

[80 rows x 10 columns]

## Exports

In [7]:
By_Country.sort_values(by = 'exp_2018', ascending=False)[['Country','exp_2018']].head(5)

Country      exp_2018
15          China  2.209089e+06
21        Germany  1.463158e+06
77            USA  1.280713e+06
42          Japan  6.586532e+05
46  Rep. of Korea  5.522279e+05

In [8]:
By_Country.sort_values(by = 'exp_2020'  , ascending=False)[['Country','exp_2020']].head(5)

Country      exp_2020
15        China  2.162897e+06
21      Germany  1.299857e+06
77          USA  1.099495e+06
42        Japan  5.680480e+05
57  Netherlands  5.119334e+05

In [18]:
By_Country[By_Country['ratio_exp_2020_2018']>1][['Country','exp_2020','exp_2018','ratio_exp_2020_2018']]

Country       exp_2020       exp_2018  ratio_exp_2020_2018
11        Brunei Darussalam    7101.852385    6218.406285             1.142070
13              Switzerland  305081.664040  295307.466841             1.033098
17                 Cameroon    4012.172120    4008.043654             1.001030
33  China and Hong Kong SAR   49330.989615   41691.934531             1.183226
38                  Ireland  176367.013568  160181.080370             1.101048
63                   Poland  228970.463180  226987.155109             1.008738
75                  Ukraine   47642.141540   47406.705905             1.004966
79                 Viet Nam  302975.611313  252105.347203             1.201782

In [27]:
41691.934531 + 0.183221*41691.934531

49330.77246770435

In [9]:
By_Country.sort_values(by = 'ratio_exp_2020_2018', ascending=False)[['Country','ratio_exp_2020_2018']].head(5)

Country  ratio_exp_2020_2018
79                 Viet Nam             1.201782
33  China and Hong Kong SAR             1.183226
11        Brunei Darussalam             1.142070
38                  Ireland             1.101048
13              Switzerland             1.033098

Top 5 biggest exportaters in 2018 : China, Germany, USA, Japan, Korea. Nearly the same order in 2020, with Netherlands in 5th place.
We can note a slight decrease of exports in 2020 for all the contries, except for 8 of them. <br/>
Expecially, Vietnamese exports increased by 20% between 2018 and 2020. Part of this performance is linked to the skilful management of the health crisis, as by reopening its factories earlier than others, Vietnam encouraged existing exporters to reallocate some of their production to the detriment of competitors such as India or Pakistan. <br/> 
Export in China and Hong Kong increase by 18% between 2018 and 2020. It's a direct impact of covid with export of hundreds fans, mask and more than two billion tests.



## Imports

In [28]:
By_Country.sort_values(by = 'imp_2018', ascending=False)[['Country','imp_2018']].head(5)

Country      imp_2018
77             USA  2.274608e+06
15           China  1.454140e+06
21         Germany  1.143163e+06
29          France  6.403833e+05
31  United Kingdom  6.028754e+05

In [29]:
By_Country.sort_values(by = 'imp_2020'  , ascending=False)[['Country','imp_2020']].head(5)

Country      imp_2020
77             USA  2.083720e+06
15           China  1.424088e+06
21         Germany  1.055009e+06
31  United Kingdom  5.637842e+05
29          France  5.404935e+05

In [34]:
By_Country[By_Country['ratio_imp_2020_2018']>1][['Country','imp_2020','imp_2018','ratio_imp_2020_2018']]

Country       imp_2020       imp_2018  ratio_imp_2020_2018
13    Switzerland  252475.809098  243753.976480             1.035781
16  Côte d'Ivoire   10614.508977    9888.893549             1.073377
17       Cameroon    7246.857730    5980.319778             1.211784
43     Kazakhstan   40768.303364   38311.806161             1.064119
56        Nigeria   52177.437775   43270.871698             1.205833
67   Saudi Arabia  112285.725517  108619.351756             1.033754
79       Viet Nam  263081.129386  242558.012896             1.084611

Top 5 biggest importer in 2018 : USA, China, Germany, France, United Kingdom. Nearly the same order for 2020. We can note a slight decrease of importation in 2020 for all the contries, except for 7 of them.

## Trade balance 

In [36]:
By_Country.sort_values(by = 'trade_balance_2018' )[['Country','trade_balance_2018']]

Country  trade_balance_2018
77                      USA      -993895.141399
31           United Kingdom      -151513.863668
33  China and Hong Kong SAR      -145737.057983
37                    India      -130458.057858
29                   France      -110660.658426
..                      ...                 ...
46            Rep. of Korea       126444.592579
67             Saudi Arabia       127439.601962
66       Russian Federation       184758.313962
21                  Germany       319995.506255
15                    China       754948.989544

[80 rows x 2 columns]

In [37]:
By_Country.sort_values(by = 'trade_balance_2020' )[['Country','trade_balance_2020']]

Country  trade_balance_2020
77                      USA      -984225.498531
31           United Kingdom      -223625.605302
29                   France       -91859.395835
33  China and Hong Kong SAR       -91185.459645
37                    India       -66673.927956
..                      ...                 ...
66       Russian Federation        93681.688519
52                   Mexico        98920.318636
55                 Malaysia       102083.952293
21                  Germany       244847.550443
15                    China       738809.102445

[80 rows x 2 columns]

In [50]:
By_Country[By_Country.Country == 'China'].exp_2020/By_Country.exp_2020.sum() *100

15    15.874386
Name: exp_2020, dtype: float64

In [38]:
By_Country.sort_values(by = 'ratio_balance_2020_2018')[['Country','ratio_balance_2020_2018']]

Country  ratio_balance_2020_2018
71         Sweden                -6.808945
16  Côte d'Ivoire                -4.036174
1       Argentina                -2.791201
56        Nigeria                -1.358428
59    New Zealand                -0.361990
..            ...                      ...
36      Indonesia                 1.957019
23        Algeria                 1.995937
79       Viet Nam                 4.178599
14          Chile                 4.526154
24        Ecuador                 7.258959

[80 rows x 2 columns]

Unsurprisingly, the trade balance with higher deficits is the United-States (2018 and 2020). The two most excess balances are China and Germany. Chinese exports represent 15% of world exports. <br/>
We can see that France is at the 5th place of the most deficit country in 2018. In 2020, France is at the 3rd place. The covid crisis has affected two of our strong points in terms of foreign trade: aeronautics and tourism. Moreover, 
the essential goods for fighting Covid-19 (masks, gloves, respirators, etc.) also weakened the French trade balance in 2020.

## References :
https://www.tresor.economie.gouv.fr/Articles/4da1ab77-66f5-4a9a-a772-0383148a4dfb/files/5144b177-f4f5-46e8-9ed7-fb0f2599c19c

https://www.tresor.economie.gouv.fr/PagesInternationales/Pages/fc06507b-ede4-4a0a-8ad0-1029f1b4db24/files/95c5dac4-07f5-4d04-a13a-a45ce36392fa

https://www.capital.fr/entreprises-marches/la-france-affiche-le-pire-deficit-commercial-de-son-histoire-1427847

https://www.tresor.economie.gouv.fr/Articles/4da1ab77-66f5-4a9a-a772-0383148a4dfb/files/5144b177-f4f5-46e8-9ed7-fb0f2599c19c

https://www.france24.com/fr/%C3%A9co-tech/20210115-la-pand%C3%A9mie-de-covid-19-dope-les-affaires-des-exportateurs-chinois